tests/components/FilterSearch.test.tsx

```diff
@@ -0,0 +1,254 @@
+
+  it('fills the search bar with an autocomplete result when a user selects it', async () => {
+    render (<FilterSearch searchFields={searchFieldsProp}/>);
+
+    const searchBarElement = screen.getByRole('textbox');
+    userEvent.type(searchBarElement, 'n');
+
+    await waitFor(() => screen.findByText('first name 1'));
+
+    userEvent.type(searchBarElement, '{arrowdown}');
+    expect(searchBarElement).toHaveValue('first name 1');
+
+    userEvent.type(searchBarElement, '{arrowdown}');
+    expect(searchBarElement).toHaveValue('first name 2');
+  });
+
+  it('triggers a search on pressing "enter" when an autocomplete result is selected', async () => {
+    const mockExecuteSearch = jest.spyOn(searchOperations, 'executeSearch').mockImplementation();
+
+    render (<FilterSearch searchFields={searchFieldsProp}/>);
+
+    const searchBarElement = screen.getByRole('textbox');
+    userEvent.type(searchBarElement, 'n');
+
+    await waitFor(() => screen.findByText('first name 1'));
+
+    userEvent.type(searchBarElement, '{arrowdown}');
+    expect(searchBarElement).toHaveValue('first name 1');
+
+    userEvent.type(searchBarElement, '{enter}');
+    const expectedSetFilterOptionParam = {
+      fieldId: 'name',
+      matcher: Matcher.Equals,
+      value: 'first name 1' ,
+      displayName: 'first name 1',
+      selected: true
+    };
+    const expectedSetOffsetParam = 0;
+    expect(setFilterOption).toBeCalledWith(expectedSetFilterOptionParam);
+    expect(setOffset).toBeCalledWith(expectedSetOffsetParam);
+
+    const setFilterOptionCallOrder = setFilterOption.mock.invocationCallOrder[0];
+    const setOffsetCallOrder = setOffset.mock.invocationCallOrder[0];
+    const mockExecuteSearchCallOrder = mockExecuteSearch.mock.invocationCallOrder[0];
+    expect(setFilterOptionCallOrder).toBeLessThan(setOffsetCallOrder);
+    expect(setOffsetCallOrder).toBeLessThan(mockExecuteSearchCallOrder);
+  });
+
+  it('does not trigger a search on pressing "enter" if no autocomplete result is selected', async () => {
+    const mockExecuteSearch = jest.spyOn(searchOperations, 'executeSearch').mockImplementation();
+    render (<FilterSearch searchFields={searchFieldsProp}/>);
+
+    const searchBarElement = screen.getByRole('textbox');
+    userEvent.type(searchBarElement, 'n{enter}');
+
+    expect(setFilterOption).not.toBeCalled();
+    expect(setOffset).not.toBeCalled();
+    expect(mockExecuteSearch).not.toBeCalled();
+  });
+
+  it('triggers a search when an autocomplete result is clicked', async () => {
+    const mockExecuteSearch = jest.spyOn(searchOperations, 'executeSearch').mockImplementation();
+    render (<FilterSearch searchFields={searchFieldsProp}/>);
+
+    const searchBarElement = screen.getByRole('textbox');
+    userEvent.type(searchBarElement, 'n');
+
+    const autocompleteSuggestion = await waitFor(() => screen.findByText('first name 1'));
+
+    userEvent.click(autocompleteSuggestion);
+    expect(setFilterOption).toBeCalled();
```

Comment:
I would just copy paste all the
```ts
const setFilterOptionCallOrder = setFilterOption.mock.invocationCallOrder[0];
    const setOffsetCallOrder = setOffset.mock.invocationCallOrder[0];
    const mockExecuteSearchCallOrder = mockExecuteSearch.mock.invocationCallOrder[0];
    expect(setFilterOptionCallOrder).toBeLessThan(setOffsetCallOrder);
// ...
```
stuff from 'triggers a search on pressing "enter" when an autocomplete result is selected'

Reply:
checks that corrected parameters were passed, and that both functions were called before executeSearch (but without comparison with each other)